cmd/prysmctl/db/query.go

```diff
@@ -0,0 +1,97 @@
+package db
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	bolt "go.etcd.io/bbolt"
+)
+
+var queryFlags = struct {
+	Path     string
+	Bucket   string
+	KeysOnly bool
+	Prefix   string
+}{}
+
+var queryCmd = &cli.Command{
+	Name:   "query",
+	Usage:  "database query tool",
+	Action: queryAction,
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:        "bucket",
+			Usage:       "boltdb bucket to search",
+			Destination: &queryFlags.Bucket,
+		},
+		&cli.StringFlag{
+			Name:        "path",
+			Usage:       "path to directory containing beaconchain.db",
+			Destination: &queryFlags.Path,
+		},
+		&cli.StringFlag{
+			Name:        "prefix",
+			Usage:       "prefix of db record key to match against (eg 0xa1 would match 0xa10, 0xa1f etc)",
+			Destination: &queryFlags.Prefix,
+		},
+		&cli.BoolFlag{
+			Name:        "print-keys",
+			Usage:       "only display keys, not values",
+			Destination: &queryFlags.KeysOnly,
+		},
+	},
+}
+
+func queryAction(_ *cli.Context) error {
+	flags := queryFlags
+	db, err := getDB(flags.Path)
+	if err != nil {
+		return err
+	}
+	if flags.Prefix != "" {
+		return prefixScan(db, flags.Bucket, flags.Prefix, flags.KeysOnly)
+	}
+	return nil
+}
+
+func prefixScan(db *bolt.DB, bucket, prefix string, keysOnly bool) error {
+	if !keysOnly {
+		return errors.New("prefix scan with value display not implemented")
+	}
```

Comment:
Wait, doesn't this mean that you have to set `print-keys`? In that case why have the flag at all? We can always add it later when value display gets implemented.

Reply:
I backed into the flags from how I would want a fuller implementation to look and then just implemented the minimum functionality to be useful to people trying to reverse partial hashes from logs. This is also the case for your other feedback -- this PR is just a sketch so we have a branch that people can grab in an emergency with the merge looming. But later it could be good to finish implementing this :)